scripts/biomedical/diseaseOntology/format_disease_ontology.py

```diff
@@ -0,0 +1,267 @@
+    """
+    sharded_list = []
+    for i in range(0, len(list_to_shard), shard_size):
+        shard = list_to_shard[i:i + shard_size]
+        arr = np.array(shard)
+        sharded_list.append(arr)
+    return sharded_list
+
+
+def col_string(df):
+    """
+    Adds string quotes to columns in a dataframe
+    Args:
+        df = dataframe whose columns are modified
+    Returns:
+        None
+    """
+    col_add = list(df['A'].unique())
+    for newcol in col_add:
+        df[newcol] = str(newcol) + ":" + df[newcol].astype(str)
+        col_rep = str(newcol) + ":" + "nan"
+        df[newcol] = df[newcol].replace(col_rep, np.nan)
+    col_names = [
+        'hasAlternativeId', 'IAO_0000115', 'hasExactSynonym', 'label', 'ICDO',
+        'MESH', 'NCI', 'SNOMEDCTUS20200901', 'UMLSCUI', 'ICD10CM', 'ICD9CM',
+        'SNOMEDCTUS20200301', 'ORDO', 'SNOMEDCTUS20180301', 'GARD', 'OMIM',
+        'EFO', 'KEGG', 'MEDDRA', 'SNOMEDCTUS20190901'
+    ]
+    for col in col_names:
+        df.update('"' + df[[col]].astype(str) + '"')
+
+
+def mesh_query(df):
+    """
+    Queries the MESH ids present in the dataframe,
+    on datacommons, fetches their dcids and adds
+    it to the same column. 
+    Args:
+        df = dataframe to change
+    Returns
+        df = modified dataframe with MESH dcid added
+    """
+    df_temp = df[df.MESH.notnull()]
+    list_mesh = list(df_temp['MESH'])
+    arr_mesh = shard(list_mesh, 1000)
+    for i in range(len(arr_mesh)):
+        query_str = """
+        SELECT DISTINCT ?id ?element_name
+        WHERE {{
+        ?element typeOf MeSHDescriptor .
+        ?element dcid ?id .
+        ?element name ?element_name .
+        ?element name {0} .
+        }}
+        """.format(arr_mesh[i])
+        result = dc.query(query_str)
+        result_df = pd.DataFrame(result)
+        result_df.columns = ['id', 'element_name']
+        df.MESH.update(df.MESH.map(result_df.set_index('element_name').id))
+    return df
+
+
+def icd10_query(df):
+    """
+    Queries the ICD10 ids present in the dataframe,
+    on datacommons, fetches their dcids and adds
+    it to the same column. 
+    Args:
+        df = dataframe to change
+    Returns
+        df = modified dataframe with ICD dcid added
+    """
+    df_temp = df[df.ICD10CM.notnull()]
+    list_icd10 = "ICD10/" + df_temp['ICD10CM'].astype(str)
+    arr_icd10 = shard(list_icd10, 1000)
+    for i in range(len(arr_icd10)):
+        query_str = """
+        SELECT DISTINCT ?id 
+        WHERE {{
+        ?element typeOf ICD10Code .
+        ?element dcid ?id .
+        ?element dcid {0} .
+        }}
+        """.format(arr_icd10[i])
+        result1 = dc.query(query_str)
+        result1_df = pd.DataFrame(result1)
+        result1_df['element'] = result1_df['?id'].str.split(pat="/").str[1]
+        result1_df.columns = ['id', 'element']
+        df.ICD10CM.update(df.ICD10CM.map(result1_df.set_index('element').id))
+    return df
+
+
+def main():
+    file_input = sys.argv[1]
```

Comment:
too much is stored under main. Put into other functions and then call in main

Reply:
Done!